### ISSUE TITLE
Flaky test: TestRunGroupAdd can fail after TestPullImageWithAllTagFromCentralRegistry

### ISSUE THREAD

**Issue by Snorch** [docker/docker#18529]
If run TestRunGroupAdd alone, busybox image is:
busybox             latest              8c2e06607696        7 months ago        2.43 MB
and it has dbus group in /etc/group

But busybox latest image pulled from docker-hub is:
busybox             latest              ac6a7980c6c2        14 hours ago        1.113 MB
and it doesn't have dbus group.

Running along with another tests(after TestPullImageWithAllTagFromCentralRegistry) busybox image changes and it is from central registry, not from tar archive and thus TestRunGroupAdd fails.

Run just those two tests on fedora 20(docker branch v1.8.3 f4bf5c70268, added MYTEST prefix to show the problem more clearly):
```
sudo TESTFLAGS='-race -test.v -check.f DockerSuite.TestMYTEST*' make test-integration-cli
...
=== RUN Test
PASS: docker_cli_pull_test.go:137: DockerSuite.TestMYTESTPullImageWithAllTagFromCentralRegistry	144.357s

----------------------------------------------------------------------
FAIL: docker_cli_run_test.go:747: DockerSuite.TestMYTESTRunGroupAdd

docker_cli_run_test.go:749:
    out, _ := dockerCmd(c, "run", "--group-add=audio", "--group-add=dbus", "--group-add=777", "busybox", "sh", "-c", "id")
github.com/docker/docker/integration-cli/_test/_obj_test/docker_utils.go:746:
    ...open github.com/docker/docker/integration-cli/_test/_obj_test/docker_utils.go: no such file or directory
... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0xc208265b60)} ("exit status 1")
... "run --group-add=audio --group-add=dbus --group-add=777 busybox sh -c id" failed with errors: Unable to find group dbus
Error response from daemon: Cannot start container 3400da8e5beb4cb0214923be98035744121e1422fdd2666a0cf8845ca142c70c: [8] System error: Unable to find group dbus
, exit status 1

OOPS: 1 passed, 1 FAILED
```

Additional info:
```
sudo docker info
Containers: 4
Images: 115
Storage Driver: devicemapper
 Pool Name: docker-253:1-1311778-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 6.023 GB
 Data Space Total: 107.4 GB
 Data Space Available: 36.83 GB
 Metadata Space Used: 9.257 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.138 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.85 (2014-04-10)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.8-100.fc20.x86_64
Operating System: Fedora 20 (Heisenbug)
CPUs: 4
Total Memory: 15.55 GiB
Name: dhcp-10-30-28-101.sw.ru
ID: Q3EP:SQO6:I3JK:PDAS:2QAY:GWEQ:M7ZN:CEEF:OBZO:QEHR:HVJ3:ETAR
```
```
docker --version
Docker version 1.8.3, build f4bf5c7
```

**Comment by thaJeztah** [comment]
Is this still an issue on master? I think some changes were made to the tests, for example; https://github.com/docker/docker/pull/16039 and https://github.com/docker/docker/pull/18503

**Comment by Snorch** [comment]
After #16039 these tests run in different order. But I think adding separate daemon for TestPullImageWithAllTagFromCentralRegistry will protect global busybox from change. And TestRunGroupAdd should be fine. 

Running command PASS, but order changed:
sudo TESTFLAGS='-race -test.v -check.f [(DockerSuite)(DockerHubPullSuite)].TestMYTEST*' make test-integration-cli

Do you know how to run these two tests in order 1)TestPullImageWithAllTagFromCentralRegistry 2)TestRunGroupAdd to be completely sure?

**Comment by thaJeztah** [comment]
The tests should be running in alphabetical order, so perhaps try renaming them

**Comment by thaJeztah** [comment]
Haven't seen this for a while
